modeshape-jcr/src/main/java/org/modeshape/jcr/cache/document/DocumentTranslator.java

```diff
@@ -868,15 +868,22 @@ protected long insertChildren( EditableDocument document,
 
     public ChildReferences getChildReferences( WorkspaceCache cache,
                                                Document document ) {
+        Name primaryType = getPrimaryType(document);
+        NodeTypes nodeTypes = getNodeTypes(cache);
+
+        boolean isUnorderedCollection = nodeTypes != null && nodeTypes.isUnorderedCollection(primaryType);
         boolean hasChildren = document.containsField(CHILDREN);
         boolean hasFederatedSegments = document.containsField(FEDERATED_SEGMENTS);
-        if (!hasChildren && !hasFederatedSegments) {
+        if (!hasChildren && !hasFederatedSegments && !isUnorderedCollection) {
             return ImmutableChildReferences.EMPTY_CHILD_REFERENCES;
         }
+
+        if (isUnorderedCollection) {
+            // the base 'mode:unordered' node types do not contain a PRB field...
+            return document.containsField(PRB) ?  new BucketedChildReferences(document, this) : ImmutableChildReferences.EMPTY_CHILD_REFERENCES; 
```

Comment:
no, it is not.
I was trying to explain that the nodes types instances (e.g. `jcr:system/jcr:nodeTypes/mode:unorderedTinyCollection`) is at runtime, an unordered collection instance which will/can never have a `$prb` field because it's the actual node type instance (sort of like the meta-data). 
That's the only time when `isUnorderedCollection && !document.containsField(PRB)` can be `true`